src/domain/landingPage/landingPageSearch/__tests__/LandingPageSearch.test.tsx

```diff
@@ -1,33 +1,109 @@
-import { act } from '@testing-library/react';
 import { advanceTo } from 'jest-date-mock';
 import React from 'react';
-import wait from 'waait';
 
-import { NeighborhoodListDocument } from '../../../../generated/graphql';
-import { render } from '../../../../util/testUtils';
+import { KeywordListDocument } from '../../../../generated/graphql';
+import { fakeKeywords } from '../../../../util/mockDataUtils';
+import {
+  actWait,
+  configure,
+  render,
+  screen,
+  userEvent,
+} from '../../../../util/testUtils';
 import LandingPageSearch from '../LandingPageSearch';
 
+configure({ defaultHidden: true });
+
+const searchValue = 'jaz';
+const keywords = fakeKeywords(2, [
+  { name: { fi: 'Jazz' } },
+  { name: { fi: 'musiikkiklubit' } },
+]);
+const keywordsResponse = { data: { keywordList: keywords } };
+
 const mocks = [
   {
     request: {
-      query: NeighborhoodListDocument,
-    },
-    result: {
-      data: {
-        neighborhoodList: {
-          data: [],
-          meta: { count: 0, next: null, previous: null },
-        },
+      query: KeywordListDocument,
+      variables: {
+        hasUpcomingEvents: true,
+        pageSize: 5,
+        text: searchValue,
       },
     },
+    result: keywordsResponse,
   },
 ];
 
-test('LandingPageSearch should match snapshot', async () => {
-  advanceTo(new Date('2020-08-20'));
-  const { container } = render(<LandingPageSearch />, { mocks });
+test('shoud route to event search page after clicking submit button', async () => {
+  const { history } = render(<LandingPageSearch />, { mocks });
+
+  userEvent.click(screen.getByRole('button', { name: /hae/i }));
+  expect(history.location.pathname).toBe('/fi/events');
+  expect(history.location.search).toBe(``);
+});
+
+test('shoud route to event search page with correct search query after clicking submit button', async () => {
```

Comment:
shoud -> should, small typo on all of these tests :)

Reply:
Fixed